app/models/status.rb

```diff
@@ -336,22 +336,18 @@ def pins_map(status_ids, account_id)
     end
 
     def reload_stale_associations!(cached_items)
-      account_ids = []
-
-      cached_items.each do |item|
-        account_ids << item.account_id
-        account_ids << item.reblog.account_id if item.reblog?
-      end
-
-      account_ids.uniq!
+      account_ids = cached_items.each_with_object([]) do |item, arr|
+        arr << item.account_id
+        arr << item.reblog&.account_id if item.reblog?
+      end.compact.uniq
 
       return if account_ids.empty?
 
-      accounts = Account.where(id: account_ids).includes(:account_stat, :user).index_by(&:id)
+      accounts_map = Account.where(id: account_ids).includes(:account_stat, :user).index_by(&:id)
 
       cached_items.each do |item|
-        item.account = accounts[item.account_id]
-        item.reblog.account = accounts[item.reblog.account_id] if item.reblog?
+        item.account        = accounts_map[item.account_id]
+        item.reblog.account = accounts_map[item.reblog.account_id] if item.reblog? && item.reblog
```

Comment:
What about setting the contents to something like `Error: mastodon#18207` so if that occurs, it doesn't completely obscures what is going on?

Reply:
Do you mean add it to the text in the serializer? I considered the idea, but making something appear in a user-controlled field seems very wrong to me. Confusion could be caused by people spoofing the error text, plus the text would be English and it could confuse the author too. Empty post isn't pretty, but it can still be deleted like a normal post.

The ideal solution is a v2 version of all endpoints that can return statuses, and an Error entity that can appear anywhere a status might appear. This would also help us render threads with gaps... So it's a long-term idea.

I'd need to check if favourites/reblogs of such a post would be appropriately rejected. I don't remember if we reject replies to reblogs, so I'd need to check if there is anything inconsistent there too. But if everywhere uses the `reblog?` check it should be fine.

Reply:
- replying to a reblog (no matter from which codepath) reassigns the reply to the reblogged statuses, and `nil` when that is soft-deleted
- it works in a similar way for reblogs
- i think it's possible to favorite a reblog through both the API and ActivityPub